akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolGateway.scala

```diff
@@ -62,9 +62,13 @@ private[http] final class PoolGateway(gatewayRef: ActorRef, val hcps: HostConnec
    * @return a Future completed when the pool has been shutdown.
    */
   def shutdown(): Future[Done] = {
-    if (!shutdownCompletedPromise.isCompleted)
+    implicit val ec = fm.executionContext
+    val shutdownFuture = shutdownCompletedPromise.future
+    shutdownFuture.foreach { _ ⇒
       gatewayRef ! Shutdown(this, shutdownCompletedPromise)
-    shutdownCompletedPromise.future
+    }
```

Comment:
This callback function must only be registered once, now it will be added another time for every call to `shutdown()` - it should go in the constructor of `PoolGateway` rather so that it is registered only once.